pennylane/qchem/hartree_fock.py

```diff
@@ -125,17 +125,19 @@ def _scf(*args):
         charges = mol.nuclear_charges
         r = mol.coordinates
         n_electron = mol.n_electrons
+        argnum = mol.argnum
+        argbools = [i in (argnum if isinstance(argnum, list) else [argnum]) for i in range(3)]
 
-        if r.requires_grad:
+        if getattr(r, "requires_grad", False) or argbools[0]:
```

Comment:
We can simply check if `0 in argnum` if we already store `mol.argnum` as a tuple/list. You can then use it at other places as well.